cranelift/2023/cranelift-08-09.md

```diff
@@ -11,12 +11,58 @@
+    - alexcrichton will try and get to reviewing it soon
+- alexcrichton
+    - With the landing of tails call work, tests weren’t running on Arm64 on MacOS. Differences in how stack pointer are handled is what was breaking the tests, alexcrichton has a PR to use the “Z” modifier instead. fitzgen had a comment on how to test this, but sadly can’t really test on CI. QEMU checks pointer authentication stuff, but the Clif config needs a certain feature flag. Off for everything but MacOS. But capturing a native backtrace fails. Currently a bug that segfaults on some unwinding in this case. 
+        - jameysharp: the release process creates some text in the auto PR that says to check a CI result, but looks like it has not been passing for some time. Embark need to change their config. Now no Mac CI. In theory Github will eventually have an Arm64 macOS target.
+        - fitzgen: can we hack this for now with an environment variable somewhere? Could add a single test that enables signing the return address. alexcrichton will try and play around with this. 
+        - Moved Amode stuff into ISLE. 
+- avanhatt: 
+    - non-code paper updates
+- jameysharp: 
+    - Doing some work on regalloc2. Want to use parallel moves resolver to improve tail call ABI support. 
+- elliottt:
+    - Experimenting with realloc2 redundant move eliminator. Also looking at better support for minimization of test cases. 
+        - Libfuzzer has built in minimization. 
+        - Using something like creduce is blocked right now because there is no text-based format for regalloc2. 
+        - Also, changing the “arbitrary instance” might also help fuzzing be more directed. Took 3 weeks to find the bug. 
+        - Are there patterns where early on bytes from libfuzzer correspond more directly to changes in the generated program. It generates only valid code. 
+        - Curious about: anything written up on how to generate inputs such that libfuzzer is better at minimizing? Probably not. A little unfortunate that we derive values from the byte buffer. 
+        - Other option: use structure-aware mutators instead of structure-aware generators. 
+- uweigand:
+    - Looked at tail call stuff, part of their ABI where they pass arguments by indirection. Not sure how that can occur with the tail call ABIs. Question: is there formal specification of the tail call ABI and what data types it supports? fizgen answer: Wasm datatypes, basically, not the fancier Cranelift things that `cg_clif` uses. Right now: arguments prepared somewhere, copied somewhere else. Would have to do pointer updates or generate differently. Right now they hardcode for i128. Could switch to using a vector register for that case. alexcrichton: on RISC-V as well there is indirection for arguments that will need to change. 
```

Comment:
The tail_call call conv doesn't need to support struct arguments I think. Wasmtime doesn't use struct arguments and cg_clif can't really use it as that would be ABI incompatible with cg_llvm and cg_gcc which both don't have support for cranelift's version of a tail call calling convention.